Sources/StreamVideo/Utils/Timers/TimerStorage.swift

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+public protocol TimerStorage {
+    func timer(for interval: TimeInterval) -> AnyPublisher<Date, Never>
+}
+
+final class StreamTimerStorage: TimerStorage {
+    private let queue = UnfairQueue()
+    private var storage: [TimeInterval: AnyPublisher<Date, Never>] = [:]
+
+    fileprivate init() {}
+
+    public func timer(for interval: TimeInterval) -> AnyPublisher<Date, Never> {
+        queue.sync {
+            if let publisher = storage[interval] {
```

Comment:
is the interval safe enough to be the key for caching timers? E.g. I want to use different timers in different places for different purposes.